db/models/orders.js

```diff
@@ -54,8 +101,69 @@ const createOrder = async ({ creatorId, name, subtotal }) => {
   }
 };
 
+const updateOrder = async ({ id, ...fields }) => {
+  const setString = Object.keys(fields)
+    .map((field, index) => {
+      return `"${field}" = $${index + 1}`;
+    })
+    .join(", ");
+  if (setString.length === 0) {
+    return;
+  }
+  try {
+    const {
+      rows: [order],
+    } = await client.query(
+      `
+            UPDATE orders
+            SET ${setString}
+            WHERE id=${id}
+            RETURNING *;
+            `,
+      Object.values(fields)
+    );
+    return order;
+  } catch (error) {
+    console.log("Error at updateOrder", error);
+    throw error;
+  }
+};
+
+const destroyOrder = async (id) => {
+  try {
+    const {
+      rows: [order],
+    } = await client.query(
+      `
+              DELETE FROM orders
+              WHERE id = $1
+              RETURNING *;
+          `,
+      [id]
+    );
+
+    await client.query(
+      `
+          DELETE FROM order_products
+          WHERE "orderId" = $1
+          RETURNING *;
+      `,
+      [id]
+    );
+
+    return order;
+  } catch (error) {
+    console.log("Error at destroyOrder", error);
+    throw error;
+  }
+};
```

Comment:
make sure we delete from order_products first here. We gotta keep in mind what is in these tables when we delete from them. The order_products table has a column of orderId that references an order.id => if we delete the orders table first and that id suddenly disappears, the order_products table that was connected to it will now throw an error cause it's reference is messed up.

Reply:
Done.